pkg/app/ops/insightcollector/deployments.go

```diff
@@ -30,129 +27,131 @@ import (
+func (c *InsightCollector) collectDevelopmentFrequency(ctx context.Context, ds []*model.Deployment, target time.Time) error {
+	apps, projects := groupDeployments(ds)
 
 	var updateErr error
 	for id, ds := range apps {
-		if err := i.updateApplicationChunks(ctx, ds[0].ProjectId, id, ds, model.InsightMetricsKind_CHANGE_FAILURE_RATE, target); err != nil {
-			i.logger.Error("failed to update application chunks", zap.Error(err))
+		if err := c.updateApplicationChunks(ctx, ds[0].ProjectId, id, ds, model.InsightMetricsKind_DEPLOYMENT_FREQUENCY, target); err != nil {
+			c.logger.Error("failed to update application chunks", zap.Error(err))
 			updateErr = err
 		}
 	}
 	for id, ds := range projects {
-		if err := i.updateApplicationChunks(ctx, id, ds[0].ApplicationId, ds, model.InsightMetricsKind_CHANGE_FAILURE_RATE, target); err != nil {
-			i.logger.Error("failed to update application chunks", zap.Error(err))
+		if err := c.updateApplicationChunks(ctx, id, "", ds, model.InsightMetricsKind_DEPLOYMENT_FREQUENCY, target); err != nil {
+			c.logger.Error("failed to update project chunks", zap.Error(err))
 			updateErr = err
 		}
 	}
 
 	return updateErr
 }
 
-func (i *InsightCollector) collectDevelopmentFrequency(ctx context.Context, ds []*model.Deployment, target time.Time) error {
-	apps, projects := i.groupDeployments(ds)
+func (c *InsightCollector) findDeploymentsCreatedInRange(ctx context.Context, from, to int64) ([]*model.Deployment, error) {
+	const pageSize = 50
+	filters := []datastore.ListFilter{
+		{
+			Field:    "CreatedAt",
+			Operator: ">=",
+			Value:    from,
+		},
+	}
 
-	var updateErr error
-	for id, ds := range apps {
-		if err := i.updateApplicationChunks(ctx, ds[0].ProjectId, id, ds, model.InsightMetricsKind_DEPLOYMENT_FREQUENCY, target); err != nil {
-			i.logger.Error("failed to update application chunks", zap.Error(err))
-			updateErr = err
+	var deployments []*model.Deployment
+	maxCreatedAt := to
+	for {
+		d, err := c.deploymentStore.ListDeployments(ctx, datastore.ListOptions{
+			PageSize: pageSize,
+			Filters: append(filters, datastore.ListFilter{
+				Field:    "CreatedAt",
+				Operator: "<",
+				Value:    maxCreatedAt,
+			}),
+			Orders: []datastore.Order{
+				{
+					Field:     "CreatedAt",
+					Direction: datastore.Desc,
+				},
+			},
+		})
+		if err != nil {
+			return nil, err
 		}
-	}
-	for id, ds := range projects {
-		if err := i.updateApplicationChunks(ctx, id, "", ds, model.InsightMetricsKind_DEPLOYMENT_FREQUENCY, target); err != nil {
-			i.logger.Error("failed to update project chunks", zap.Error(err))
-			updateErr = err
+		if len(d) == 0 {
+			// get all deployments in range
+			break
 		}
+
+		deployments = append(deployments, d...)
+		maxCreatedAt = d[len(d)-1].CreatedAt
 	}
+	return deployments, nil
+}
 
-	return updateErr
+func (c *InsightCollector) findDeploymentsCompletedInRange(ctx context.Context, from, to int64) ([]*model.Deployment, error) {
+	const pageSize = 50
```

Comment:
should we keep it as global var/const as previously 🤔

Reply:
I have no strong opinion about this. Let me revert it.